### PR TITLE
fix: footer nav bar

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- A lot of refactor on the footer nav bar for maintainability. It was difficult to follow the ternary conditions which sometimes are only for the legacy and sometimes for the new one.
- We will proceed here only one or the other, so I instead introduced separate components to manage each variant.
- Key fixes here are:
  - Backdrop on the floating input.
  - Issue with indicator not working properly for control.
  - Support both v1 and control for indicators.

Previews:

https://github.com/dailydotdev/apps/assets/13744167/81a4d5f8-9233-4363-9897-30f0281a5c9a

Feel free to check the old behavior from production. Fixed preview:
https://github.com/dailydotdev/apps/assets/13744167/6cfc1ee9-8d1f-4c6d-b234-0a2df3cdb2cc

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

MI-256 #done

### Preview domain
https://mi-256.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/styles/base.css [hunk 1]
```diff
@@ -36,6 +36,15 @@ body {
   }
 
   overflow-x: hidden;
+
+  @screen mobileL {
```
Comment (sshanzel) [inline packages/shared/src/styles/base.css:40]: This helps fixing the footer navbar. One of the issues is that on all pages, you must add extra padding due to the fact that we are using `fixed` property. With the properties below, this will make using `sticky` work.

File: packages/webapp/components/FooterNavBar.tsx [hunk 2]
```diff
@@ -1,326 +1,104 @@
-import React, { ReactElement, useContext, useMemo, useState } from 'react';
```
Comment (sshanzel) [inline packages/webapp/components/FooterNavBar.tsx:1]: Separated into two components.

File: packages/webapp/components/footer/FooterNavBarControl.tsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
+import React, { ReactElement } from 'react';
```
Comment (sshanzel) [inline packages/webapp/components/footer/FooterNavBarControl.tsx:1]: Basically the old FooterNavBar, just broken down into multiple components.

File: packages/webapp/components/FooterNavBar.tsx [hunk 4]
```diff
@@ -1,326 +1,104 @@
...
+  home: 'Home',
+  search: 'Search',
+  bookmarks: 'Bookmarks',
+  notifications: 'Notifications',
+  squads: 'Squads',
+  profile: 'Profile',
+};
+
+const combinedTabs = tabs.concat(
+  ...(mobileUxTabs.filter((tab) => !isValidElement(tab)) as FooterTab[]),
```
Comment (DragosIorgulescu) [inline packages/webapp/components/FooterNavBar.tsx:39]: why is this needed here?

**Review by DragosIorgulescu — COMMENTED** [review]
Great work on the refactoring, I don't see anything obvious, overall looks great, just have one concern regarding the preview, it looks like the active `flipper` is delayed? 

It seems that it's always one step behind the current active tab? Eg. when navigating from search to squads, the active state moves from home to search, etc.

File: packages/webapp/components/FooterNavBar.tsx (hunk 4 above)
Comment (sshanzel) [reply packages/webapp/components/FooterNavBar.tsx:39]: This is if in case there is no active page returned by the `useActiveNav`, we will check the paths here if any matches the current route.

Comment (DragosIorgulescu) [reply packages/webapp/components/FooterNavBar.tsx:39]: no, I mean why do you need to combine tabs if we are splitting the components rendering them?

File: packages/shared/src/components/MainLayout.tsx [hunk 5]
```diff
@@ -225,7 +225,7 @@ function MainLayoutComponent({
       />
       <main
         className={classNames(
-          'flex flex-row',
+          'flex flex-col tablet:flex-row',
```
Comment (rebelchris) [inline packages/shared/src/components/MainLayout.tsx:228]: This could have massive impact on some pages as many use this.
We have to make tripple sure none break, think about devcard, sources, tags, whatever hidden page we have.

Can you just make sure before we merge (and really try to find all pages 🙏 )

File: packages/shared/src/styles/base.css (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/styles/base.css:40]: This feels very heavy to add to generic body tag 🙈 
I'm a bit worried it will have some implication on somethings we haven't tested.

File: packages/shared/src/styles/base.css [hunk 6]
```diff
@@ -661,6 +670,14 @@ html.light .invert .invert,
   position: relative;
   flex-direction: column;
   align-items: stretch;
+
+  @screen mobileL {
+    flex: 1;
+  }
+
+  @screen tablet {
+    flex: unset;
```
Comment (rebelchris) [inline packages/shared/src/styles/base.css:679]: This as well

**Review by rebelchris — COMMENTED** [review]
The split is very good, love it.
Just have some concerns on the super global CSS implications, but not sure how to address them 😅

CC: @nensidosari maybe you have some idea how we can validate impact across the board?

File: packages/shared/src/styles/base.css (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/styles/base.css:40]: Yeah, same sentiments. However, I think it is best to use `sticky` on the footer navbar as it is problematic across all pages which this fixes. Also why I am immediately setting the values back to its original state after mobile.

File: packages/webapp/components/FooterNavBar.tsx (hunk 4 above)
Comment (sshanzel) [reply packages/webapp/components/FooterNavBar.tsx:39]: We have to support it the way we define which tab should be active for indicator using the `flipper`. Otherwise, we could've done it directly from either component.

File: packages/shared/src/components/MainLayout.tsx (hunk 5 above)
Comment (sshanzel) [reply packages/shared/src/components/MainLayout.tsx:228]: Yeah, but I realized no pages are using row layout for mobile since it wouldn't fit. With that, definitely going to triple-check all pages.

File: packages/shared/src/styles/base.css (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/styles/base.css:40]: There is no way we can only target that element vs the whole body tag?

Comment (sshanzel) [reply packages/shared/src/styles/base.css:40]: Let me check, though as far as I tested yesterday, those were the requirements. Let me check if I move the elements 1 level deeper.

File: packages/shared/src/styles/base.css (hunk 6 above)
Comment (nensidosari) [reply packages/shared/src/styles/base.css:679]: This just nulls out the one set above so not much concerns on it

**Review by nensidosari — APPROVED** [review]
Love the refactoring. 

As about the css changes, a bit concerned as well, in idea they look ok since it's only changing in mobile.

Since we don't have tests 😬 , no better idea 🤷‍♀️ I think the only way to test is manually. I can help test a bunch of pages, we can split if we want to make this faster and see if we detect anything obvious...

File: packages/shared/src/styles/base.css (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/styles/base.css:40]: I checked having it inside `MainLayout` component but still needed the CSS properties defined here. With that, I've created a list to check on all 3 screen sizes.

Maybe we should store this list somewhere to make it easier next time we have something similar to this in the future.
Screenshot 2024-04-04 at 5 35 04 PM

**Comment by sshanzel** [comment]
> Since we don't have tests 😬 , no better idea 🤷‍♀️ I think the only way to test is manually. I can help test a bunch of pages, we can split if we want to make this faster and see if we detect anything obvious...

Thank you for the offer, I think I still have the time to check them myself, though I will share a list if you want to run through the same things I will be running through.

**Review by rebelchris — APPROVED** [review]
Won't block it, but will manually check your screenshots again :)
